polkadot/node/core/pvf/src/execute/queue.rs

```diff
@@ -278,15 +292,59 @@ impl Queue {
 		}
 
 		let job = queue.remove(job_index).expect("Job is just checked to be in queue; qed");
+		let exec_kind = job.exec_kind;
+
+		if !self.is_job_relevant(exec_kind) {
+			let _ = job.result_tx.send(Err(ValidationError::ExecutionDeadline));
+			gum::warn!(
+				target: LOG_TARGET,
+				?priority,
+				?exec_kind,
+				"Job exceeded its deadline and was dropped without execution",
+			);
+			return self.try_assign_next_job(finished_worker);
```

Comment:
Can we avoid recursion here ? it makes me uneasy.

Reply:
Yeah, with this new mechanics it's a good idea